arctic/serialization/numpy_records.py

```diff
@@ -47,6 +56,55 @@ def _to_primitive(arr, string_max_len=None, forced_dtype=None):
     return arr
 
 
+def treat_tz_as_dateutil(tz) -> bool:
+    """
+    Return whether the given tz object is from `dateutil`
+
+    Vendored from Pandas:
+    https://github.com/pandas-dev/pandas/blob/v1.3.5/pandas/_libs/tslibs/timezones.pyx#L66-L67
+    """
+    return hasattr(tz, '_trans_list') and hasattr(tz, '_trans_idx')
+
+
+def consistent_get_timezone_str(tz: Union[datetime.tzinfo, str]) -> str:
+    """
+    Convert a tzinfo object to a serializable string
+
+    Unlike the Pandas `get_timezone` function, this function should always return a string.
+    """
+    if isinstance(tz, str):
+        return tz
+
+    # The behaviour of Pandas' `get_timezone()` for UTC `tzinfo`s differs across versions.
+    # This is due to either changes to the underlying `is_utc()` function, or the behaviour when it returns `True`.
+    #
+    # Differing implementations:
+    #   `pandas` 0.22.0 - https://github.com/pandas-dev/pandas/blob/v0.22.0/pandas/_libs/tslibs/timezones.pyx#L71-L72
+    #      - Returns "UTC" for UTC `tzinfo`s, except:
+    #        - `dateutil.tz.gettz("UTC")` - Returns a string like "dateutil//usr/share/zoneinfo/UTC"
+    #   `pandas` 0.24.0 - https://github.com/pandas-dev/pandas/blob/v0.24.0/pandas/_libs/tslibs/timezones.pyx#L59-L60
+    #      - Returns the `tzinfo` object for UTC `tzinfo`s except:
+    #        - `dateutil.tz.gettz("UTC")` - Returns a string like "dateutil//usr/share/zoneinfo/UTC"
+    #   `pandas` 1.3.0 -  https://github.com/pandas-dev/pandas/blob/v1.3.0/pandas/_libs/tslibs/timezones.pyx#L53
+    #      - Returns the `tzinfo` object for UTC `tzinfo`s (including `dateutil.tz.gettz("UTC")`)
+    if PD_VER < "0.24.0":
+        return str(get_timezone(tz))
```

Comment:
Can we get to this point with tz==None?

Reply:
This function is only called for non-`None` `tz`s. The type hint reflects this: https://github.com/man-group/arctic/pull/1013/files#diff-1c730dd7d8c45baa8edaa77fcdd8b32c1c10c91c370a9aa46b218f59fbe07a0eR69